server/src/main/java/io/crate/execution/dml/Indexer.java

```diff
@@ -563,6 +660,11 @@ public ParsedDocument index(IndexItem item) throws IOException {
             Object[] values = item.insertValues();
             for (int i = 0; i < values.length; i++) {
                 Reference reference = columns.get(i);
+                if (targetsHaveOids && reference instanceof DynamicReference == false) {
+                    // It's possible to have references with OID after doing a mapping update:
```

Comment:
maybe better name for the flag would be `targetsSupportOids`?
otherwise looks contradicting that we say targets have but still need to exclude dynamicRefs